src/libraries/System.IO.MemoryMappedFiles/ref/System.IO.MemoryMappedFiles.cs

```diff
@@ -24,6 +24,8 @@ namespace System.IO.MemoryMappedFiles
     public partial class MemoryMappedFile : System.IDisposable
     {
         internal MemoryMappedFile() { }
+        public MemoryMappedFile(Microsoft.Win32.SafeHandles.SafeMemoryMappedFileHandle handle) { }
+        public MemoryMappedFile(Microsoft.Win32.SafeHandles.SafeMemoryMappedFileHandle handle, System.IDisposable? fileStream, bool leaveOpen) { }
```

Comment:
Need to bring this new constructor to API review.

Reply:
I don't know that `leaveOpen` is useful, or needed... Generally it's used because the disposable parameter is being used as its strong type (e.g. data being read from a Stream).  For this constructor, it seems like if you would pass false, just pass null for the parent (currently named fileStream, but that's not a great name, since it can be anything)

Reply:
I shared your initial suggestion to @JeremyKuhne about passing `null` and removing `leaveOpen`.

The reason why we kept the boolean is because we need to root the file stream to the `MemoryMappedFile`. We can't pass `null`. The boolean will tell the constructor if the passed disposable object should be disposed or not.

When the user decides the object should be left open, then we pass a disposable wrapper class that will root the fileStream inside it.

Reply:
> When the user decides the object should be left open, then we pass a disposable wrapper class that will root the fileStream inside it.

Right, so you don't need the `leaveOpen` parameter on the constructor :smile:.  You can stay rooted by passing an IDisposable wrapper that just doesn't forward the call to Dispose.

Reply:
🤓 Gah, you're right. I'll fix that.

Reply:
While the wrapper workaround works, I'm concerned about having this take an `IDisposable`. Users would be forced to create the wrapper as well when they want to not have the `MemoryMappedFile` close the backing stream.  Documentation would be something like this:

| Disposable for the the `FileStream` used to open the `SafeMemoryMappedFileHandle`, if any. The disposable will be disposed when the `MemoryMappedFile` is disposed.
| Notes: If you need to keep the source `FileStream` (if any) open after the `MemoryMappedFile` is disposed, either don't pass the handle in or wrap it in an `IDisposable` wrapper class that doesn't call dispose.

@carlossanlop Can you test what happens when using the `MemoryMappedFile` after the opening `FileStream` is disposed? Given the old API surface my presumption is that the `FileStream` _has_ to be left open. I haven't validated that, however. _If_ `MemoryMappedFile` works fine with the `FileStream` closed, then I'm much less concerned about not being more generic in the API. If it breaks using the `MemoryMappedFile` then I'd be even more inclined to use `FileStream, bool` like we do everywhere else. If that can't get approved we'll need to be very explicit in the docs about rooting the `FileStream` to avoid finalizer issues.

Reply:
> _If MemoryMappedFile works fine with the FileStream closed, then I'm much less concerned about not being more generic in the API._

I added a new unit test that:

1. Creates a `FileStream`.
2. Creates a new `MemoryMappedFileStream` using the new `CreateFromFile` method, by consuming that `FileStream`.
3. Disposes the `DileStream`.
4. Creates a `MemoryMappedViewAccessor` and reads some bytes from it, then disposes it.
5. Creates a `MemoryMappedViewStream` and reads some bytes from it, then disposes it.
5. Disposes the `MemoryMappedViewAccessor`.

The unit test passed.